### PR TITLE
fix: increase/decrease radio buttons & feat: decimal frequency config

### PR DESCRIPTION
**Pull request by TonybynMp4**
## Description
They shouldn't work when the radio is off!
Also made the subfrequency check faster.

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -170,18 +169,20 @@ RegisterNUICallback('volumeDown', function(_, cb)
 end)
 
 RegisterNUICallback('increaseradiochannel', function(_, cb)
-    local newChannel = radioChannel + 1
-    exports['pma-voice']:setRadioChannel(newChannel)
-    exports.qbx_core:Notify(Lang:t('new_channel')..newChannel, 'success')
+    if not onRadio then return end
```
Comment (BerkieBb) [inline client/main.lua]: This is technically unnecessary as you can't trigger the nui callback without being on the radio

File: client/main.lua [hunk 2]
```diff
@@ -170,18 +169,20 @@ RegisterNUICallback('volumeDown', function(_, cb)
...
 
 RegisterNUICallback('decreaseradiochannel', function(_, cb)
     if not onRadio then return end
-    local newChannel = radioChannel - 1
-    if newChannel >= 1 then
-        exports['pma-voice']:setRadioChannel(newChannel)
-        exports.qbx_core:Notify(Lang:t('new_channel')..newChannel, 'success')
+    radioChannel -= 1
+    radioChannel = radioChannel < 1 and 1 or radioChannel
+    if radioChannel >= 1 then
```
Comment (BerkieBb) [inline client/main.lua]: This check is also redundant due to the ternary operator above making sure it doesn't go below 1

File: client/main.lua (hunk 1 above)
Comment (TonybynMp4) [reply client/main.lua]: even if it's powered off the nui callbacks trigger 🤔

File: client/main.lua (hunk 2 above)
Comment (TonybynMp4) [reply client/main.lua]: already fixed :)

File: client/main.lua [hunk 3]
```diff
@@ -147,44 +147,45 @@ RegisterNUICallback('leaveRadio', function(_, cb)
...
-		exports['pma-voice']:setRadioVolume(radioVolume)
-	else
-		exports.qbx_core:Notify(Lang:t('max_volume'), 'error')
+	if not onRadio then return cb('ok') end
+	if radioVolume > 95 then
+		return exports.qbx_core:Notify(Lang:t('max_volume'), 'error')
 	end
-    cb('ok')
+
+	radioVolume = radioVolume + 5
```
Comment (BerkieBb) [inline client/main.lua]: +=

File: client/main.lua [hunk 4]
```diff
@@ -147,44 +147,45 @@ RegisterNUICallback('leaveRadio', function(_, cb)
...
-		exports.qbx_core:Notify(Lang:t('new_volume')..radioVolume, 'success')
-		exports['pma-voice']:setRadioVolume(radioVolume)
-	else
+	if not onRadio then return cb('ok') end
+	if radioVolume < 10 then
 		exports.qbx_core:Notify(Lang:t('min_volume'), 'error')
 	end
-    cb('ok')
+
+	radioVolume = radioVolume - 5
```
Comment (BerkieBb) [inline client/main.lua]: -=

File: client/main.lua (hunk 1 above)
Comment (BerkieBb) [reply client/main.lua]: That shouldn't be possible though aa that requires user input to change the channel

Comment (TonybynMp4) [reply client/main.lua]: the UI can be open without the radio being active 🤔

Comment (BerkieBb) [reply client/main.lua]: Ah yeah, nvm then

File: client/main.lua [hunk 5]
```diff
@@ -103,14 +102,14 @@ RegisterNetEvent('qbx_radio:client:use', function()
...
-RegisterNetEvent('qbx_radio:client"onRadioDrop', function()
+RegisterNetEvent('qbx_radio:client:onRadioDrop', function()
     if radioChannel ~= 0 then
         leaveradio()
     end
 end)
 
 RegisterNUICallback('joinRadio', function(data, cb)
-    local rchannel = tonumber(data.channel)
+    local rchannel = config.decimalPlaces and math.floor(tonumber(data.channel) * (10 ^ config.decimalPlaces)) or tonumber(data.channel)
```
Comment (BerkieBb) [inline client/main.lua]: Could've just done `qbx.math.round(tonumber(data.channel), config.decimalPlaces)`

**Review by BerkieBb — APPROVED** [review]

File: client/main.lua (hunk 5 above)
Comment (TonybynMp4) [reply client/main.lua]: oh yeah lol

**Comment by BerkieBb** [comment]
Simple thing you can do as well is instead of making config.decimalPlaces a boolean or number, just make it a number and use that to let them decide if it should be a whole number or a decimal number, then you can remove the ternary statement and just do `local channel = qbx.math.round(tonumber(data.channel), config.decimalPlaces)`

**Comment by TonybynMp4** [comment]
> Simple thing you can do as well is instead of making config.decimalPlaces a boolean or number, just make it a number and use that to let them decide if it should be a whole number or a decimal number, then you can remove the ternary statement and just do `local channel = qbx.math.round(tonumber(data.channel), config.decimalPlaces)`

I was trying to keep the current behavior, but if you prefer that i can add it real quick

**Comment by BerkieBb** [comment]
> > Simple thing you can do as well is instead of making config.decimalPlaces a boolean or number, just make it a number and use that to let them decide if it should be a whole number or a decimal number, then you can remove the ternary statement and just do `local channel = qbx.math.round(tonumber(data.channel), config.decimalPlaces)`
> 
> I was trying to keep the current behavior, but if you prefer that i can add it real quick

It wouldn't change behaviour, just what the user inputs

**Comment by TonybynMp4** [comment]
> It wouldn't change behaviour, just what the user inputs

Currently a player can add as many decimal places as they want so it would, but i guess it doesn't really matter much, sounds like something qbus overlooked.

**Comment by BerkieBb** [comment]
> > It wouldn't change behaviour, just what the user inputs
> 
> Currently a player can add as many decimal places as they want so it would, but i guess it doesn't really matter much, sounds like something qbus overlooked.

Ahh that's what you mean, yeah let the server owner/dev limit it

File: config/shared.lua [hunk 6]
```diff
@@ -1,5 +1,12 @@
 return {
     maxFrequency = 500, -- Max amount of available channel frequencies to use
+    
+    -- If false restrictedChannels restricts all decimals, if true you need to manually add each subchannel (100.01, 100.02 etc)
+    whitelistSubChannels = false,
+
+    ---@type number
+    -- How many decimal places to use for the channel.
```
Comment (solareon) [inline config/shared.lua]:
```suggestion
    -- How many decimal places to use for the subchannel.
```

**Review by solareon — APPROVED** [review]

**Pull request merged**